test/lightning/workflows_test.exs

```diff
@@ -0,0 +1,64 @@
+defmodule Lightning.WorkflowsTest do
+  use Lightning.DataCase, async: true
+
+  alias Lightning.Workflows
+  alias Lightning.Workflows.Workflow
+
+  import Lightning.WorkflowsFixtures
+
+  describe "workflows" do
+    test "list_workflows/0 returns all workflows" do
+      workflow = workflow_fixture()
+      assert Workflows.list_workflows() == [workflow]
+    end
+
+    test "get_workflow!/1 returns the workflow with given id" do
+      workflow = workflow_fixture()
+      assert Workflows.get_workflow!(workflow.id) == workflow
+
+      assert_raise Ecto.NoResultsError, fn ->
+        Workflows.get_workflow!(Ecto.UUID.generate())
+      end
+    end
+
+    test "get_workflow/1 returns the workflow with given id" do
+      assert Workflows.get_workflow(Ecto.UUID.generate()) == nil
+
+      workflow = workflow_fixture()
+      assert Workflows.get_workflow(workflow.id) == workflow
+    end
+
+    test "create_workflow/1 with valid data creates a workflow" do
+      valid_attrs = %{name: "some-name"}
+
+      assert {:ok, %Workflow{} = workflow} =
+               Workflows.create_workflow(valid_attrs)
+
+      assert workflow.name == "some-name"
+    end
+
+    test "update_workflow/2 with valid data updates the workflow" do
```

Comment:
once uniqueness is added, show that you can't name two workflows the same thing if they're in the same project

Reply:
Done